### PR TITLE
fix(nav flow): hardware back button is closing modals when needed

### PR DESCRIPTION
**Pull request by pvinis**
The type of this PR is: **Bugfix**

This PR resolves [CX-1191] [CX-1190]

### Description

Fixed this with @ds300 and @brainbicycle during bugbash.

When we show a "confirmation" screen, like bid success or registration success, then the android hw back button should take us all the way back, by just dismissing the modal.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-434]: https://artsyproduct.atlassian.net/browse/CX-434
[CX-1191]: https://artsyproduct.atlassian.net/browse/CX-1191
[CX-1190]: https://artsyproduct.atlassian.net/browse/CX-1190

### CODE REVIEW

File: src/lib/Components/Bidding/Screens/BidResult.tsx [hunk 1]
```diff
@@ -45,6 +45,18 @@ const Icons = {
 }
 
 export class BidResult extends React.Component<BidResultProps> {
+  backButtonListener?: NativeEventSubscription = undefined
+
+  componentDidMount = () => {
+    this.backButtonListener = BackHandler.addEventListener("hardwareBackPress", this.handleBackButton.bind(this))
```
Comment (ds300) [inline src/lib/Components/Bidding/Screens/BidResult.tsx:51]: nonblocking

```suggestion
    this.backButtonListener = BackHandler.addEventListener("hardwareBackPress", this.handleBackButton)
```

it's already bound

File: src/lib/Components/Bidding/Screens/BidResult.tsx [hunk 2]
```diff
@@ -45,6 +45,18 @@ const Icons = {
...
+  }
+
+  componentWillUnmount = () => {
+    if (this.backButtonListener) {
+      this.backButtonListener.remove()
+    }
```
Comment (ds300) [inline src/lib/Components/Bidding/Screens/BidResult.tsx:57]: nonblocking
```suggestion
    this.backButtonListener?.remove()
```

File: src/lib/Components/Bidding/Screens/RegistrationResult.tsx [hunk 3]
```diff
@@ -106,6 +106,30 @@ const resultEnumToPageName = (result: RegistrationStatus) => {
     } as any) /* STRICTNESS_MIGRATION */
 )
 export class RegistrationResult extends React.Component<RegistrationResultProps> {
+  backButtonListener?: NativeEventSubscription = undefined
+
+  componentDidMount = () => {
+    this.backButtonListener = BackHandler.addEventListener("hardwareBackPress", this.handleBackButton.bind(this))
+  }
```
Comment (ds300) [inline src/lib/Components/Bidding/Screens/RegistrationResult.tsx:113]: nonblocking
```suggestion
  componentDidMount = () => {
    this.backButtonListener = BackHandler.addEventListener("hardwareBackPress", this.handleBackButton)
  }
```

File: src/lib/Components/Bidding/Screens/RegistrationResult.tsx [hunk 4]
```diff
@@ -106,6 +106,30 @@ const resultEnumToPageName = (result: RegistrationStatus) => {
...
+
+  componentWillUnmount = () => {
+    if (this.backButtonListener) {
+      this.backButtonListener.remove()
+    }
```
Comment (ds300) [inline src/lib/Components/Bidding/Screens/RegistrationResult.tsx:118]:
```suggestion
    this.backButtonListener?.remove()
```

**Review by ds300 — APPROVED** [review]
Looks good. I spotted some nitpick syntax stuff, probably not worth fixing, feel free to ignore.

File: src/lib/Components/Bidding/Screens/BidResult.tsx (hunk 2 above)
Comment (pvinis) [reply src/lib/Components/Bidding/Screens/BidResult.tsx:57]: I appreciate the direct code diff suggestion!

**Pull request merged**